tests/unit/test_routing_fixes.py

```
@@ -0,0 +1,96 @@
+        if os.path.exists(test_file):
+            with open(test_file, 'r') as f:
+                content = f.read()
+            namespace_matches = re.findall(r"Namespace\('([^']*)'", content)
+            for match in namespace_matches:
+                self.assertFalse(match.startswith('/'), f"Found leading slash in namespace '{match}' in {test_file}")
+
+    def test_root_endpoints_before_api_init_in_test_files(self):
+        """Test that test files have root endpoints registered before Flask-RESTX init."""
+        # Test one file at a time to avoid loops in tests
+        test_file = PROJECT_ROOT / 'deployment' / 'cloud-run' / 'test_swagger_debug.py'
+        if os.path.exists(test_file):
```

Ruff (0.12.2)

`os.path.exists()` should be replaced by `Path.exists()`

(PTH110)